common/src/main/java/org/apache/celeborn/common/util/PbSerDeUtils.java

```diff
@@ -90,18 +90,35 @@ public static FileInfo fromPbFileInfo(PbFileInfo pbFileInfo)
         userIdentifier);
   }
 
+  public static FileInfo fromPbFileInfo(PbFileInfo pbFileInfo, UserIdentifier userIdentifier) {
+    return new FileInfo(
+        pbFileInfo.getFilePath(),
+        new ArrayList<>(pbFileInfo.getChunkOffsetsList()),
+        userIdentifier);
+  }
+
   public static PbFileInfo toPbFileInfo(FileInfo fileInfo) {
     PbFileInfo.Builder builder = PbFileInfo.newBuilder();
     builder.setFilePath(fileInfo.getFilePath()).addAllChunkOffsets(fileInfo.getChunkOffsets());
     return builder.build();
   }
 
-  public static ConcurrentHashMap<String, FileInfo> fromPbFileInfoMap(byte[] data)
+  public static ConcurrentHashMap<String, FileInfo> fromPbFileInfoMap(
+      byte[] data, ConcurrentHashMap<String, UserIdentifier> cache)
       throws InvalidProtocolBufferException {
     PbFileInfoMap pbFileInfoMap = PbFileInfoMap.parseFrom(data);
     ConcurrentHashMap<String, FileInfo> fileInfoMap = new ConcurrentHashMap<>();
     for (Map.Entry<String, PbFileInfo> entry : pbFileInfoMap.getValuesMap().entrySet()) {
-      fileInfoMap.put(entry.getKey(), fromPbFileInfo(entry.getValue()));
+      FileInfo fileInfo;
+      PbUserIdentifier pbUserIdentifier = entry.getValue().getUserIdentifier();
+      String userIdentifierKey = pbUserIdentifier.getTenantId() + "-" + pbUserIdentifier.getName();
```

Comment:
why not just ConcurrentHashMap<PbUserIdentifier, UserIdentifier>

Reply:
Does PbUserIdentifier override equals?

Reply:
Seems PbUserIdentifier indeed overrides equals, but I think it's fine to use String here since PbUserIdentifier's equals seems a little bit complex.
```
  public boolean equals(final java.lang.Object obj) {
    if (obj == this) {
     return true;
    }
    if (!(obj instanceof org.apache.celeborn.common.protocol.PbUserIdentifier)) {
      return super.equals(obj);
    }
    org.apache.celeborn.common.protocol.PbUserIdentifier other = (org.apache.celeborn.common.protocol.PbUserIdentifier) obj;

    if (!getTenantId()
        .equals(other.getTenantId())) return false;
    if (!getName()
        .equals(other.getName())) return false;
    if (!unknownFields.equals(other.unknownFields)) return false;
    return true;
  }
```

Reply:
Yes it overwrites both equals and hashCode method in the PbUserIdentifier, I was worry about the some unknown rare cases mentioned in the first answer of https://stackoverflow.com/questions/34729968/protobuf-objects-as-keys-in-maps . But it seems it won't happen in PbUserIdentifier, and even there is an added optional field of PbUserIdentifier which might cause some unknown issue in future, I think it might not have much impact, it would just create one more object.
I did some test and seems this rare case mentioned in stackoverflow still exists in proto3.
<img width="814" alt="image" src="https://user-images.githubusercontent.com/30563796/196207881-a65c3d72-4153-481a-9394-96d9374cc06f.png">
There is some test i did with the modified PbUserIdentifier(add a optional string field `testCol`)
<img width="963" alt="image" src="https://user-images.githubusercontent.com/30563796/196210523-ad9e56b8-a886-4731-80ff-a7f8106dcbfa.png">